src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Navigation/NavigationExtensions.uwp.cs

```diff
@@ -33,6 +36,24 @@ Page GetCurrentPage(Page currentPage)
 			}
 		}
 
+		/// <summary>
+		/// ATTENTION: Create the Renderer for UWP Don't use the one Provided by Xamarin.Forms, Causes a crash in Native Compiled Code
+		/// 1. DefaultRenderer is PopupRenderer instead of DefaultRenderer()
+		/// 2. No Invalid Cast Exceptions in UWP Native when the Xamarin Forms Renderer Functions is used.
+		/// </summary>
+		/// <param name="element">Element for getting the renderer</param>
+		/// <returns>Renderer</returns>
+		// https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.UAP/Platform.cs
+		public static IVisualElementRenderer CreateRenderer(VisualElement element)
+		{
+			if (element == null)
+				throw new ArgumentNullException(nameof(element));
+
+			var renderer = Registrar.Registered.GetHandlerForObject<IVisualElementRenderer>(element) ?? new PopupRenderer();
+
+			return renderer;
```

Comment:
How about setting the element here similar to Forms implementation?

I mean ```CreateRenderer``` will both create renderer and set element
Then it can return void, I guess

Reply:
Good idea